pkg/ytconfig/generator_test.go

```diff
@@ -14,61 +15,180 @@ import (
 )
 
 var (
+	testClusterDomain             = "fake.zone"
 	testLogRotationPeriod   int64 = 900000
 	testTotalLogSize              = 10 * int64(1<<30)
 	testMasterExternalHosts       = []string{
 		"host1.external.address",
 		"host2.external.address",
 		"host3.external.address",
 	}
+	testBasicInstanceSpec = v1.InstanceSpec{InstanceCount: 3}
+	testStorageClassname  = "yc-network-hdd"
+	testResourceReqs      = corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU:    *resource.NewQuantity(20, resource.BinarySI),
+			corev1.ResourceMemory: *resource.NewQuantity(5*1024*1024*1024, resource.BinarySI),
+		},
+	}
+	testLocationChunkStore = v1.LocationSpec{
+		LocationType: "ChunkStore",
+		Path:         "/yt/hdd1/chunk-store",
+		Medium:       "nvme",
+	}
+	testLocationChunkCache = v1.LocationSpec{
+		LocationType: "ChunkCache",
+		Path:         "/yt/hdd1/chunk-cache",
+	}
+	testLocationSlots = v1.LocationSpec{
+		LocationType: "Slots",
+		Path:         "/yt/hdd2/slots",
+	}
+	testVolumeMounts = []corev1.VolumeMount{
+		{
+			Name:      "hdd1",
+			MountPath: "/yt/hdd1",
+		},
+	}
+	testVolumeClaimTemplates = []v1.EmbeddedPersistentVolumeClaim{
+		{
+			EmbeddedObjectMetadata: v1.EmbeddedObjectMetadata{
+				Name: "hdd1",
+			},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+				StorageClassName: &testStorageClassname,
+			},
+		},
+	}
+	testClusterNodeSpec = v1.ClusterNodesSpec{
+		Tags: []string{"rack:xn-a"},
+		Rack: "fake",
+	}
 )
 
+func TestGetChytInitClusterConfig(t *testing.T) {
+	g := NewGenerator(getYtsaurus(), testClusterDomain)
+	cfg, err := g.GetChytInitClusterConfig()
+	require.NoError(t, err)
+	canonize.Assert(t, cfg)
+}
+func TestGetControllerAgentsConfig(t *testing.T) {
+	g := NewGenerator(withControllerAgents(getYtsaurus()), testClusterDomain)
+	cfg, err := g.GetControllerAgentConfig()
+	require.NoError(t, err)
+	canonize.Assert(t, cfg)
+}
+func TestGetDataNodeConfig(t *testing.T) {
+	g := NewGenerator(getYtsaurus(), testClusterDomain)
+	cfg, err := g.GetDataNodeConfig(getDataNodeSpec())
+	require.NoError(t, err)
+	canonize.Assert(t, cfg)
+}
+func TestGetDiscoveryConfig(t *testing.T) {
+	g := NewGenerator(withDiscovery(getYtsaurus()), testClusterDomain)
+	cfg, err := g.GetDiscoveryConfig()
+	require.NoError(t, err)
+	canonize.Assert(t, cfg)
+}
+func TestGetExecNodeConfig(t *testing.T) {
+	g := NewGenerator(getYtsaurus(), testClusterDomain)
```

Comment:
Maybe add exec nodes in ytsaurus spec?
Same for data nodes

Reply:
As we agreed in chat, I've introduced a new function "ytsaurus spec with every component" and used it in all non-specific tests, so we test generating configs for more realistic spec. 
And for tests that want to test something specific (like master hostnames) we can build spec granularly adding required functions.